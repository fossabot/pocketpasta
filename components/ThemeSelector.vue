<template>
  <form
    @submit.prevent="
      $auth.$storage.setUniversal('theme', $store.state.theme, true)
    "
  >
    <h1>Settings</h1>
    <div class="form-group">
      <label for="theme">Select a theme</label>
      <select
        id="theme"
        v-model="currentTheme"
        name="theme"
        class="form-control"
      >
        <option disabled value="">Please select a theme</option>
        <option
          v-for="theme in $store.state.themes"
          :key="theme.value"
          :value="theme.value"
          class="form-control"
        >
          {{ theme.value }}
        </option>
      </select>
    </div>
    <button class="btn btn-primary">
      Save theme
    </button>
  </form>
</template>

<script>
export default {
  computed: {
    currentTheme: {
      get() {
        return this.$store.state.theme.value;
      },
      set(value) {
        this.$store.commit('setTheme', this.$store.getters.getThemeById(value));
      },
    },
  },
  head() {
    return {
      link: this.$store.state.themes.map((theme) => ({
        hid: `theme-${theme.value}`,
        disabled: 'disabled',
        href: theme.href,
        rel: 'preload',
        as: 'style',
      })),
    };
  },
};
</script>
