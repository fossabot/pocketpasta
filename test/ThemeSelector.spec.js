import { shallowMount, createLocalVue } from '@vue/test-utils';
import ThemeSelector from '@/components/ThemeSelector.vue';
import Vuex from 'vuex';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('ThemeSelector', () => {
  let getters;
  let state;
  let store;
  let mutations;

  beforeEach(() => {
    state = {
      theme: {
        name: 'darky',
        href: 'https://bootswatch.com/4/darkly/bootstrap.min.css',
      },
    };

    getters = {
      clicks: () => 2,
      inputValue: () => 'input',
    };

    mutations = {
      setTheme(currentState, theme) {
        // const newState = { ...currentState };
        currentState.theme = theme;
      },
    };

    store = new Vuex.Store({
      state,
      getters,
      mutations,
    });
  });

  const factory = () =>
    shallowMount(ThemeSelector, {
      store,
      localVue,
    });

  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('theme getter', () => {
    const wrapper = factory();
    wrapper.vm.currentTheme = 'darkly';
    expect(wrapper.vm.currentTheme).toBeTruthy();
  });
});
