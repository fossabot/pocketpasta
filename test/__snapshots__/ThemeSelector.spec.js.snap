// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSelector renders properly 1`] = `
"<form><h1>Settings</h1> <div class=\\"form-group\\"><label for=\\"theme\\">Select a theme</label> <select id=\\"theme\\" name=\\"theme\\" class=\\"form-control\\"><option disabled=\\"disabled\\" value=\\"\\">Please select a theme</option> <option class=\\"form-control\\" value=\\"cerulean\\">
        cerulean
      </option><option class=\\"form-control\\" value=\\"cosmo\\">
        cosmo
      </option><option class=\\"form-control\\" value=\\"cyborg\\">
        cyborg
      </option><option class=\\"form-control\\" value=\\"darkly\\">
        darkly
      </option><option class=\\"form-control\\" value=\\"flatly\\">
        flatly
      </option><option class=\\"form-control\\" value=\\"journal\\">
        journal
      </option><option class=\\"form-control\\" value=\\"litera\\">
        litera
      </option><option class=\\"form-control\\" value=\\"lumen\\">
        lumen
      </option><option class=\\"form-control\\" value=\\"lux\\">
        lux
      </option><option class=\\"form-control\\" value=\\"materia\\">
        materia
      </option><option class=\\"form-control\\" value=\\"minty\\">
        minty
      </option><option class=\\"form-control\\" value=\\"pulse\\">
        pulse
      </option><option class=\\"form-control\\" value=\\"sandstone\\">
        sandstone
      </option><option class=\\"form-control\\" value=\\"simplex\\">
        simplex
      </option><option class=\\"form-control\\" value=\\"sketchy\\">
        sketchy
      </option><option class=\\"form-control\\" value=\\"slate\\">
        slate
      </option><option class=\\"form-control\\" value=\\"solar\\">
        solar
      </option><option class=\\"form-control\\" value=\\"spacelab\\">
        spacelab
      </option><option class=\\"form-control\\" value=\\"superhero\\">
        superhero
      </option><option class=\\"form-control\\" value=\\"untied\\">
        untied
      </option><option class=\\"form-control\\" value=\\"yeti\\">
        yeti
      </option></select></div> <button class=\\"btn btn-primary\\">
    Save theme
  </button></form>"
`;
